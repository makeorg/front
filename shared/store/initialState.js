// @flow
import { type StateRoot } from './types';

export const initialState: StateRoot = {
  appConfig: {
    source: '',
    language: '',
    country: '',
    translations: {},
    queryParams: {},
  },
  proposal: {
    hasProposed: false,
    popularProposals: [],
    error: undefined,
    data: undefined,
  },
  sequence: {
    isSequenceCollapsed: false,
    firstProposal: undefined,
    questionSlug: undefined,
    currentIndex: 0,
    votedProposalIds: {},
    proposals: [],
  },
  questions: {},
  currentQuestion: '',
  notification: {
    contentType: undefined,
  },
  user: {
    authentication: {
      errors: [],
      isLoggedIn: false,
      user: undefined,
    },
    passwordRecovery: {
      newPassword: undefined,
      resetToken: undefined,
      userId: undefined,
      errorMessage: undefined,
      error: false,
      updated: false,
    },
  },
  modal: {
    isOpen: false,
    contentType: '',
    showExpirationSession: false,
  },
  partners: {},
};

export const createInitialState = (): StateRoot =>
  JSON.parse(JSON.stringify(initialState));
