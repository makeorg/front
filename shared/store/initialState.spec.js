import { initialState } from './initialState';

it('initialState snapshot', () => {
  expect(initialState).toMatchSnapshot();
});
