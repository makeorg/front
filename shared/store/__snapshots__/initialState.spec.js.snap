// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialState snapshot 1`] = `
Object {
  "appConfig": Object {
    "countriesWithConsultations": Array [],
    "country": "",
    "language": "",
    "queryParams": Object {},
    "source": "",
    "translations": Object {},
  },
  "currentQuestion": "",
  "modal": Object {
    "contentType": "",
    "focusAfterClose": true,
    "isOpen": false,
    "showExpirationSession": false,
  },
  "notification": Object {
    "contentType": undefined,
  },
  "panel": Object {
    "isExpanded": false,
    "panelContent": undefined,
  },
  "partners": Object {},
  "proposal": Object {
    "data": undefined,
    "error": undefined,
    "hasProposed": false,
    "popularProposals": Array [],
  },
  "questions": Object {},
  "sequence": Object {
    "cards": Array [],
    "currentIndex": 0,
    "firstProposal": undefined,
    "isSequenceCollapsed": false,
    "proposals": Array [],
    "questionSlug": undefined,
    "votedProposalIds": Object {},
  },
  "session": Object {
    "sessionId": "",
  },
  "user": Object {
    "authentication": Object {
      "errors": Array [],
      "isLoggedIn": false,
      "user": undefined,
    },
    "passwordRecovery": Object {
      "error": false,
      "errorMessage": undefined,
      "newPassword": undefined,
      "resetToken": undefined,
      "updated": false,
      "userId": undefined,
    },
  },
  "views": Object {},
}
`;
